src/dune_engine/rules.ml

```diff
@@ -28,7 +28,7 @@ module Dir_rules = struct
 
   type t = data Id.Map.t
 
-  let data_to_dyn = function
```

Comment:
Should we document this convention in hacking.rst?

Reply:
Sure, it can't hurt.